### PR TITLE
refactor: Change Dockerfile and improve CI/CD pipeline

### PR DESCRIPTION
**Pull request by ponkio-o**
Hi @dag-andersen,

Thank you for developing a great tool. I did refactor `Dockerfile` and CI/CD pipeline.

### Change to use `TARGETARCH` variable to build multi-arch container image
`TARGETARCH` is embedded default variable of buidkit.
https://docs.docker.com/reference/dockerfile/#automatic-platform-args-in-the-global-scope

This variable useful when you build multi-arch container image with single `Dockerfile`. I did change to use this variable and use single Dockerfile.

### Change to use `debian:bookworm-slim` for base image of argocd-diff-preview
`dagandersen/argocd-diff-preview` is very heavy.  The current size of this image is 1.53GB.
These are affected for CI performance when using containerized argocd-diff-preview. Therefore, I did change to use debian slim image for final build stage.

```console
$ docker images | grep dagandersen/argocd-diff-preview 
dagandersen/argocd-diff-preview                                                                     v0.0.22                                    921cbb2000e0   3 days ago       1.53GB
```

After the change, the image size is now about 500MB.
```console
$ docker images | grep use-debian-slim-0.0.22-
argocd-diff-preview                                                                                 use-debian-slim-0.0.22-amd                 f3ef9eba2ebf   32 minutes ago   517MB
argocd-diff-preview                                                                                 use-debian-slim-0.0.22-arm                 5fa99f77cb6a   39 minutes ago   534MB
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+COPY --from=build /argocd-diff-preview/kind /usr/local/bin/kind
+COPY --from=build /argocd-diff-preview/kubectl /usr/local/bin/kubectl
+COPY --from=build /usr/local/bin/helm /usr/local/bin/helm
+COPY --from=build /usr/local/bin/argocd /usr/local/bin/argocd
+COPY --from=build /argocd-diff-preview/target/release/argocd-diff-preview .
+
+RUN apt-get update && \
+    apt-get install -y git && \
+    apt-get clean && \
```
Comment (ponkio-o) [inline Dockerfile:55]:
```suggestion
```

**Review by dag-andersen — CHANGES_REQUESTED** [review]
Let's also fix the `makefile` [link](https://github.com/ponkio-o/argocd-diff-preview/blob/c1afbe2d68fc0e9a5644cf74b67c9e4285899fba/makefile#L25) by deleting _line 4_ + deleting the `-f $(docker_file)` 🚀

**Comment by dag-andersen** [comment]
Amazing!

The smaller image is great! That part has annoyed me for a long time.

I tried to get `TARGETARCH` to work a few months ago, but I gave up because, to me, all the Docker multi-arch commands and documentation were a mess 😅

_side note:_
Overall, I'm not very happy with the release process and the automation around the tool, so this should definitely be improved over time. I’d also like to implement some caching for the images so it doesn't take so long, but I just haven’t gotten around to it.

File: .github/workflows/pr-build.yml [hunk 2]
```diff
@@ -9,8 +9,12 @@ on:
       - Cargo.toml
```
[the remark this answers is not in the document]
Comment (ponkio-o) [reply .github/workflows/pr-build.yml:9]: Thank you! I applied your suggestion the following commit.
https://github.com/dag-andersen/argocd-diff-preview/pull/56/commits/5efe8be3e615b865fa4abea8ec3718eb4cfb52df

**Comment by ponkio-o** [comment]
> Overall, I'm not very happy with the release process and the automation around the tool, so this should definitely be improved over time.

I see. Actually, I was thinking the same thing. I believe the following points can be improved. If you agree I will work on these improvemets.

- Cache optimization in docker build
- Renovate for automatic updating of dependencies (Actions, CI-related tools, cargo...etc)

**Comment by dag-andersen** [comment]
That would be great! 🚀 

Other items on the roadmap that you might find interesting:

- Support for pre-releases. The current release pipeline does not support creating "pre-releases" that I can test before doing a final release. Having a nicer flow for testing binaries would be great. 

- Improved testing capabilities.  
    - Currently, I test most features by running commands in the Makefile and manually validating the output. I also rebase eight open PRs to the main branch and verify they still work and produce the correct output. Ideally, I’d like a comprehensive `make test` command that runs and verifies the output across various test branches.

- Running the tool without needing access to the host’s Docker daemon.  
   - Currently, the tool relies on the host machine’s Docker daemon to create a kind cluster. Working with Docker-in-Docker can be challenging, so I thought it might be better to run the kind cluster with Podman [[link]](https://kind.sigs.k8s.io/docs/user/rootless/) inside the Docker `argocd-diff-preview` container. This way, everything could operate within the same container, eliminating the need for `--network=host -v /var/run/docker.sock:/var/run/docker.sock`.

**Review by dag-andersen — APPROVED** [review]

**Pull request merged**